xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -63,6 +72,48 @@ bool IsBatchedMatmul(const HloInstruction* instr) {
   return is_batch_dot;
 }
 
+// We need to check if current gemm is sharing a parent node with a forward
+// fMHA call. We check this by doing a BFS of all operands to see if there's
+// any user that is a forward fMHA custom call.
+bool IsSharingOperandWithFwdMha(HloInstruction* gemm) {
+  for (int64_t i = 0; i < gemm->operands().size(); i++) {
+    std::queue<HloInstruction*> visit_list;
+    visit_list.push(gemm->mutable_operand(i));
+    while (!visit_list.empty()) {
+      HloInstruction* current_instr = visit_list.front();
+      for (int64_t user_index = 0; user_index < current_instr->user_count();
+           user_index++) {
+        HloInstruction* user = current_instr->users()[user_index];
+        switch (user->opcode()) {
+          case HloOpcode::kBitcast:
+          case HloOpcode::kReshape:
+          case HloOpcode::kTranspose: {
```

Comment:
Could you update the function doc to indicate you're looking for a chain of bitcast/reshape/transpose?

Reply:
added doc to reflect this